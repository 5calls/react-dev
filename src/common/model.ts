
export interface Issue {
  id: string;
  name: string;
  reason: string;
  script: string;
  contacts?: Contact[];
  contactType?: string;
  categories: Category[];
  inactive: boolean;
  outcomeModels: Outcome[];
  link: string;
  linkTitle: string;
  slug: string;
}

export const DefaultIssue: Issue = {
  id: '',
  name: '',
  reason: '',
  script: '',
  categories: [],
  inactive: false,
  outcomeModels: [],
  link: '',
  linkTitle: '',
  slug: '',
};

export interface Outcome {
  label: string;
  status: string;
}

export interface Contact {
  id: string;
  name: string;
  phone: string;
  photoURL?: string;
  party: Party;
  state: string;
  reason: string;
  area?: string;
  field_offices?: FieldOffice[];
}

export interface VoterContact {
  id: string;
  name: string;
  city: string;
  phone: string;
}

export interface Category {
  name: string;
}

export class CategoryMap {
  category: Category;
  issues: Issue[];
}

export type Party = 'Democrat' | 'Republican' | 'Independent' | '';

// export const DefaultContact: Contact = {
//   name: '',
//   phone: '',
//   party: '',
//   state: '',
//   reason: ''
// };

export const DefaultContact: Contact = {} as Contact;

export interface FieldOffice {
  city: string;
  phone: string;
}

export interface UserStat {
  all: string[];
  contactedCount: number;
  voiceMailCount: number;
  unavailableCount: number;
}

export interface Group {
  id: string;
  name: string;
  description: string;
  totalCalls: number;
}

/**
 * Represents the place used to get location.
 * It may be one of three options:
 * 1. CACHED_ADDRESS - address stored in local storage
 * 2. BROWSER_GEOLOCATION - address obtained from the browser geolocation API
 * 3. IP_INFO - address obtained from ipinfo.io API
 */
export enum LocationFetchType {
  CACHED_ADDRESS = 'CACHED_ADDRESS', // 'address' in Choo version
  BROWSER_GEOLOCATION = 'BROWSER_GEOLOCATION', // 'browserGeolocation' in Choo version
  IP_INFO = 'IP_INFO' // 'ipAddress' in Choo version
}

/**
 * Encapsulates a location using latitude
 * and longitude. Undefined for either longitude
 * or latitude indicates that the geolocation has
 * has not been set.
 */
export interface GeolocationPosition {
  longitude: number | undefined;
  latitude: number | undefined;
}

/* 5 Calls API data */
export interface ApiData {
  splitDistrict: boolean;
  invalidAddress: boolean;
  normalizedLocation: string | undefined;
  divisions: string[];
  issues: Issue[];
}

export interface GroupIssues {
  splitDistrict: boolean;
  invalidAddress: boolean;
  normalizedLocation: string | undefined;
  divisions: string[];
  issues: Issue[];
}

export interface CountData {
  count: number; // total call count
}

/* Data from iponfo.io API */
export interface IpInfoData {
  ip: string;
  hostname: string;
  city: string;
  region: string; // state
  country: string;
  loc: string; // long, lat - used in issue lookup
  org: string; // internet service provider
  postal: string; // zip code
}

export enum LocationUiState {
  FETCHING_LOCATION = 'FETCHING_LOCATION',
  LOCATION_FOUND = 'LOCATION_FOUND',
  ENTERING_LOCATION = 'ENTERING_LOCATION',
  LOCATION_ERROR = 'LOCATION_ERROR'
}

export interface DonationGoal {
  goal: Donations; //
}

export interface Donations {
    count: number; // number of donors
    amount: number; // total collected (api===amount)
    total: number; // goal (api===total)
    kind: string; // denomincation (dollars)
}
