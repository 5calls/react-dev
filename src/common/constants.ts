
export const APP_NAME = '5 Calls';
export const APP_URL = 'https://5calls.org';
// export const APP_URL = 'http://localhost:8090';
export const API_URL = 'https://api.5calls.org/v1';
export const ISSUES_API_URL = `${APP_URL}/issues/?all=true&address=`;
export const REPORT_API_URL = `${APP_URL}/report`;
export const COUNTS_API_URL = `${API_URL}/counts`;
export const DONATIONS_API_URL = `${API_URL}/donations`;
export const GROUP_API_URL = `${API_URL}/groups`;
export const IP_INFO_URL = 'https://ipinfo.io/json';

export const zipCodeRegex: RegExp = /^\d{5}(-\d{4})?$/;

export const contact = {
  email: 'make5calls@gmail.com',
  github: 'https://github.com/5calls',
  twitter: 'https://twitter.com/make5calls',
  facebook: 'https://www.facebook.com/make5calls',
  apps: 'https://crgj.app.link/7R2bEB0R4F',
};
