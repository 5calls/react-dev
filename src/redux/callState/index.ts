export {
  CallStateAction, CallStateActionType,
  CurrentIssueAction, NextContact,
  SetContactIdsAction, SetShowFieldOfficeNumbers,
  CompleteIssueAction, ClearContactIndexesAction,
  JoinGroupAction } from './action';
export {completeIssueActionCreator, moveToNextActionCreator,
  selectIssueActionCreator, clearContactIndexes, joinGroupActionCreator } from './actionCreator';
export { callStateReducer, CallState } from './reducer';
export { OutcomeData, FlexibleOutcomeData, submitOutcome, submitFlexibleOutcome } from './asyncActionCreator';
