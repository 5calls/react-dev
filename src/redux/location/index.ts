export { LocationAction, LocationActionType, LocationClearedAction,
  LocationSetAction, InvalidAddressAction, SetUiStateAction,
  CacheCityAction, FetchingLocationAction, ValidatingLocationAction } from './action';
export { setLocation, clearAddress, setInvalidAddress, setCachedCity,
  setFetchingLocation, setValidatingLocation, setUiState} from './actionCreator';
export { LocationState, locationStateReducer } from './reducer';
export { setAddress } from './asyncActionCreator';
