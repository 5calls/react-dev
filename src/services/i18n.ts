import * as i18n from 'i18next';
import * as XHR from 'i18next-xhr-backend';
// import Cache from 'i18next-localstorage-cache';
import * as LanguageDetector from 'i18next-browser-languagedetector';

const instance = i18n
    .use(XHR)
    // .use(Cache)
    .use(LanguageDetector)
    .init({
        backend: {
            loadPath: '/locales/{{lng}}.json'
        },
        fallbackLng: 'en',

        react: {
            wait: true, // globally set to wait for loaded translations in translate hoc
            // exposeNamespace: true // exposes namespace on data-i18next-options to be used in eg. locize-editor
        },

        // have a common namespace used around the full app
        ns: [''],
        defaultNS: '',

        debug: true,

        // cache: {
        //   enabled: true
        // },

        interpolation: {
            escapeValue: false, // not needed for react!!
            formatSeparator: ','
        }
    });

export default instance;
