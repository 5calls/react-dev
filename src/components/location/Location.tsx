import * as React from 'react';
import { LocationState } from '../../redux/location/reducer';
import { TranslationFunction } from 'i18next';
import { translate } from 'react-i18next';
import { LocationUiState } from '../../common/model';

interface Props {
  readonly locationState: LocationState;
  readonly setLocation: (location: string) => void;
  readonly clearLocation: () => void;
  readonly t: TranslationFunction;
}

// this will be needed
interface State {
  location: string;
  uiState: LocationUiState;
}

export class Location extends React.Component<Props, State> {

  private addressInput: HTMLInputElement | null;

  constructor(props: Props) {
    super(props);
    // set initial state
    this.state = this.setStateFromProps(props);
  }

  componentWillReceiveProps(nextProps: Props) {
    this.setState(this.setStateFromProps(nextProps));
  }

  componentDidMount() {
    if (this.addressInput) {
      this.addressInput.focus();
    }
  }

  /**
   * Set state from props when props
   * are initialized or refreshed
   *
   * @param {Props} props
   * @returns {State}
   */
  setStateFromProps(props: Props): State {
    let location = props.locationState.cachedCity || props.locationState.address ;
    let uiState  = props.locationState.uiState ;
    if (!location  && uiState !== LocationUiState.ENTERING_LOCATION) {
      uiState = LocationUiState.LOCATION_ERROR;
    }

    return {
      location,
      uiState
    };
  }

  getWidgetTitle() {
    let title = <span/>;
    switch (this.state.uiState) {
      case LocationUiState.LOCATION_FOUND:
        title = <p id="locationMessage">{this.props.t('location.yourLocation')}: <span>{this.state.location}</span></p>;
        break;
      case LocationUiState.FETCHING_LOCATION:
        title = <p id="locationMessage" className="loadingAnimation">{this.props.t('location.gettingYourLocation')}</p>;
        break;
      case LocationUiState.LOCATION_ERROR:
        title = <p id="locationMessage" role="alert">{this.props.t('location.invalidAddress')}</p>;
        break;
      case LocationUiState.ENTERING_LOCATION:
        title = <p id="locationMessage">{this.props.t('location.chooseALocation')}</p>;
        break;
      default:
        break;
    }
    return title;
  }

  getWidget() {
    let widget = <span/>;
    switch (this.state.uiState) {
      case LocationUiState.FETCHING_LOCATION:
        // No widget in this case
        break;
      case LocationUiState.LOCATION_FOUND:
        const enterLocation = (e) => {
          e.preventDefault();
          this.props.clearLocation();
        };
        widget = <div><button onClick={enterLocation}>{this.props.t('location.changeLocation')}</button></div>;
        break;
      case LocationUiState.LOCATION_ERROR:
      // FIXME: clear address text box
        // console.log('Clear address text box')
        // console.log('Remove break statement')
        // break;
      case LocationUiState.ENTERING_LOCATION:
        const submitAddress = (e) => {
          e.preventDefault();
          const newLocation = e.target.elements.address.value;
          this.props.setLocation(newLocation);
        };
        const clearTextBox = (e) => { e.target.value = ''; };
        widget = (
          <div>
            <form onSubmit={submitAddress} >
              <input
                type="text"
                ref={(input) => { this.addressInput = input; }}
                autoFocus={true}
                id="address"
                name="address"
                aria-labelledby="locationMessage"
                aria-invalid={this.state.uiState === LocationUiState.LOCATION_ERROR}
                onFocus={clearTextBox}
                placeholder="Enter an address or zip code"
              />
              <button>{this.props.t('common.go')}</button>
            </form>
          </div>
          );
        break;
      default:
        break;
    }
    return widget;
  }

  render() {
    return (
      <div>
        {this.getWidgetTitle()}
        {this.getWidget()}
      </div>
    );
  }
}

export const LocationTranslatable = translate()(Location);
