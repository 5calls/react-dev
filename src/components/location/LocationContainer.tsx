import { connect, Dispatch } from 'react-redux';
import { bindActionCreators } from 'redux';
import { setAddress, clearAddress } from '../../redux/location';
import { ApplicationState } from '../../redux/root';
import { Location } from './index';

interface StateProps {
  readonly location: string;
  readonly isValid: boolean;
  readonly isLoading: boolean;
}

interface DispatchProps {
  readonly setLocation: (location: string) => void;
  readonly clearLocation: () => void;
}

function mapStateToProps(state: ApplicationState): StateProps {
  return {
    location: state.locationState.address || state.locationState.cachedCity,
    isValid: !state.locationState.invalidAddress,
    isLoading: state.locationState.fetchingLocation || state.locationState.validatingLocation
  };
}

function mapDispatchToProps(dispatch: Dispatch<DispatchProps>): DispatchProps {
  return bindActionCreators({setLocation: setAddress, clearLocation: clearAddress}, dispatch);
}

export default connect<StateProps, DispatchProps, {}>(
    mapStateToProps, mapDispatchToProps)(Location);
