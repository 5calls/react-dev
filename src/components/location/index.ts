import { Location, LocationTranslatable } from './Location';
import LocationContainer from './LocationContainer';

export { Location, LocationTranslatable, LocationContainer };
