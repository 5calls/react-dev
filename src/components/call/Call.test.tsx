import * as React from 'react';
import { shallow } from 'enzyme';
import i18n from '../../services/i18n';
import { Call } from './index';
import { Issue, DefaultIssue, LocationUiState, LocationFetchType } from '../../common/model';
import { CallState } from '../../redux/callState';
import { LocationState } from '../../redux/location';

test('Call component should be rendered if passed a valid object', () => {
  const issue: Issue = Object.assign({}, DefaultIssue, { id: '1', name: 'testName' });
  let callState: CallState = {
    currentIssueId: 'test1',
    contactIndexes: {'test1': 2, 'test2': 1},
    completedIssueIds: ['test1', 'test2'],
    group: undefined,
  };
  const locationState: LocationState = {
    address: '1234',
    cachedCity: 'Anytown',
    splitDistrict: false,
    uiState: LocationUiState.LOCATION_FOUND,
    locationFetchType: LocationFetchType.BROWSER_GEOLOCATION
  };
  const onSubmitOutcome = jest.fn();
  const clearLocation = jest.fn();
  const component = shallow(
    <Call
      issue={issue}
      callState={callState}
      splitDistrict={false}
      clearLocation={clearLocation}
      locationState={locationState}
      t={i18n.t}
      onSubmitOutcome={onSubmitOutcome}
    />);
  expect(component).toMatchSnapshot();
});