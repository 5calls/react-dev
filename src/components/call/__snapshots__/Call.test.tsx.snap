// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Call component should be rendered if passed a valid object 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <section
    className="call"
>
    <Translate(Component)
        currentIssue={
            Object {
                "categories": Array [],
                "id": "1",
                "inactive": false,
                "link": "",
                "linkTitle": "",
                "name": "testName",
                "outcomeModels": Array [],
                "reason": "",
                "script": "",
              }
        }
        t={[Function]}
    />
    <Translate(ContactDetails)
        contactIndex={0}
        currentIssue={
            Object {
                "categories": Array [],
                "id": "1",
                "inactive": false,
                "link": "",
                "linkTitle": "",
                "name": "testName",
                "outcomeModels": Array [],
                "reason": "",
                "script": "",
              }
        }
        t={[Function]}
    />
    <Translate(Component)
        contactIndex={0}
        issue={
            Object {
                "categories": Array [],
                "id": "1",
                "inactive": false,
                "link": "",
                "linkTitle": "",
                "name": "testName",
                "outcomeModels": Array [],
                "reason": "",
                "script": "",
              }
        }
        locationState={
            Object {
                "address": "1234",
                "cachedCity": "Anytown",
                "locationFetchType": "BROWSER_GEOLOCATION",
                "splitDistrict": false,
                "uiState": "LOCATION_FOUND",
              }
        }
        t={[Function]}
    />
    <Translate(withRouter(Outcomes))
        currentContactId=""
        currentIssue={
            Object {
                "categories": Array [],
                "id": "1",
                "inactive": false,
                "link": "",
                "linkTitle": "",
                "name": "testName",
                "outcomeModels": Array [],
                "reason": "",
                "script": "",
              }
        }
        numberContactsLeft={0}
        onSubmitOutcome={[Function]}
        t={[Function]}
    />
    
</section>,
  "nodes": Array [
    <section
      className="call"
>
      <Translate(Component)
            currentIssue={
                  Object {
                        "categories": Array [],
                        "id": "1",
                        "inactive": false,
                        "link": "",
                        "linkTitle": "",
                        "name": "testName",
                        "outcomeModels": Array [],
                        "reason": "",
                        "script": "",
                      }
            }
            t={[Function]}
      />
      <Translate(ContactDetails)
            contactIndex={0}
            currentIssue={
                  Object {
                        "categories": Array [],
                        "id": "1",
                        "inactive": false,
                        "link": "",
                        "linkTitle": "",
                        "name": "testName",
                        "outcomeModels": Array [],
                        "reason": "",
                        "script": "",
                      }
            }
            t={[Function]}
      />
      <Translate(Component)
            contactIndex={0}
            issue={
                  Object {
                        "categories": Array [],
                        "id": "1",
                        "inactive": false,
                        "link": "",
                        "linkTitle": "",
                        "name": "testName",
                        "outcomeModels": Array [],
                        "reason": "",
                        "script": "",
                      }
            }
            locationState={
                  Object {
                        "address": "1234",
                        "cachedCity": "Anytown",
                        "locationFetchType": "BROWSER_GEOLOCATION",
                        "splitDistrict": false,
                        "uiState": "LOCATION_FOUND",
                      }
            }
            t={[Function]}
      />
      <Translate(withRouter(Outcomes))
            currentContactId=""
            currentIssue={
                  Object {
                        "categories": Array [],
                        "id": "1",
                        "inactive": false,
                        "link": "",
                        "linkTitle": "",
                        "name": "testName",
                        "outcomeModels": Array [],
                        "reason": "",
                        "script": "",
                      }
            }
            numberContactsLeft={0}
            onSubmitOutcome={[Function]}
            t={[Function]}
      />
      
</section>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Call
        callState={
                Object {
                        "completedIssueIds": Array [
                          "test1",
                          "test2",
                        ],
                        "contactIndexes": Object {
                          "test1": 2,
                          "test2": 1,
                        },
                        "currentIssueId": "test1",
                        "group": undefined,
                      }
        }
        clearLocation={[Function]}
        issue={
                Object {
                        "categories": Array [],
                        "id": "1",
                        "inactive": false,
                        "link": "",
                        "linkTitle": "",
                        "name": "testName",
                        "outcomeModels": Array [],
                        "reason": "",
                        "script": "",
                      }
        }
        locationState={
                Object {
                        "address": "1234",
                        "cachedCity": "Anytown",
                        "locationFetchType": "BROWSER_GEOLOCATION",
                        "splitDistrict": false,
                        "uiState": "LOCATION_FOUND",
                      }
        }
        onSubmitOutcome={[Function]}
        splitDistrict={false}
        t={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Call {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "callState": Object {
            "completedIssueIds": Array [
              "test1",
              "test2",
            ],
            "contactIndexes": Object {
              "test1": 2,
              "test2": 1,
            },
            "currentIssueId": "test1",
            "group": undefined,
          },
          "clearLocation": [Function],
          "issue": Object {
            "categories": Array [],
            "id": "1",
            "inactive": false,
            "link": "",
            "linkTitle": "",
            "name": "testName",
            "outcomeModels": Array [],
            "reason": "",
            "script": "",
          },
          "locationState": Object {
            "address": "1234",
            "cachedCity": "Anytown",
            "locationFetchType": "BROWSER_GEOLOCATION",
            "splitDistrict": false,
            "uiState": "LOCATION_FOUND",
          },
          "onSubmitOutcome": [Function],
          "splitDistrict": false,
          "t": [Function],
        },
        "refs": Object {},
        "state": Object {
          "currentContact": undefined,
          "currentContactIndex": 0,
          "issue": Object {
            "categories": Array [],
            "id": "1",
            "inactive": false,
            "link": "",
            "linkTitle": "",
            "name": "testName",
            "outcomeModels": Array [],
            "reason": "",
            "script": "",
          },
          "numberContactsLeft": 0,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <section
          className="call"
>
          <Translate(Component)
                    currentIssue={
                              Object {
                                        "categories": Array [],
                                        "id": "1",
                                        "inactive": false,
                                        "link": "",
                                        "linkTitle": "",
                                        "name": "testName",
                                        "outcomeModels": Array [],
                                        "reason": "",
                                        "script": "",
                                      }
                    }
                    t={[Function]}
          />
          <Translate(ContactDetails)
                    contactIndex={0}
                    currentIssue={
                              Object {
                                        "categories": Array [],
                                        "id": "1",
                                        "inactive": false,
                                        "link": "",
                                        "linkTitle": "",
                                        "name": "testName",
                                        "outcomeModels": Array [],
                                        "reason": "",
                                        "script": "",
                                      }
                    }
                    t={[Function]}
          />
          <Translate(Component)
                    contactIndex={0}
                    issue={
                              Object {
                                        "categories": Array [],
                                        "id": "1",
                                        "inactive": false,
                                        "link": "",
                                        "linkTitle": "",
                                        "name": "testName",
                                        "outcomeModels": Array [],
                                        "reason": "",
                                        "script": "",
                                      }
                    }
                    locationState={
                              Object {
                                        "address": "1234",
                                        "cachedCity": "Anytown",
                                        "locationFetchType": "BROWSER_GEOLOCATION",
                                        "splitDistrict": false,
                                        "uiState": "LOCATION_FOUND",
                                      }
                    }
                    t={[Function]}
          />
          <Translate(withRouter(Outcomes))
                    currentContactId=""
                    currentIssue={
                              Object {
                                        "categories": Array [],
                                        "id": "1",
                                        "inactive": false,
                                        "link": "",
                                        "linkTitle": "",
                                        "name": "testName",
                                        "outcomeModels": Array [],
                                        "reason": "",
                                        "script": "",
                                      }
                    }
                    numberContactsLeft={0}
                    onSubmitOutcome={[Function]}
                    t={[Function]}
          />
          
</section>,
        "_debugID": 2,
        "_renderedOutput": <section
          className="call"
>
          <Translate(Component)
                    currentIssue={
                              Object {
                                        "categories": Array [],
                                        "id": "1",
                                        "inactive": false,
                                        "link": "",
                                        "linkTitle": "",
                                        "name": "testName",
                                        "outcomeModels": Array [],
                                        "reason": "",
                                        "script": "",
                                      }
                    }
                    t={[Function]}
          />
          <Translate(ContactDetails)
                    contactIndex={0}
                    currentIssue={
                              Object {
                                        "categories": Array [],
                                        "id": "1",
                                        "inactive": false,
                                        "link": "",
                                        "linkTitle": "",
                                        "name": "testName",
                                        "outcomeModels": Array [],
                                        "reason": "",
                                        "script": "",
                                      }
                    }
                    t={[Function]}
          />
          <Translate(Component)
                    contactIndex={0}
                    issue={
                              Object {
                                        "categories": Array [],
                                        "id": "1",
                                        "inactive": false,
                                        "link": "",
                                        "linkTitle": "",
                                        "name": "testName",
                                        "outcomeModels": Array [],
                                        "reason": "",
                                        "script": "",
                                      }
                    }
                    locationState={
                              Object {
                                        "address": "1234",
                                        "cachedCity": "Anytown",
                                        "locationFetchType": "BROWSER_GEOLOCATION",
                                        "splitDistrict": false,
                                        "uiState": "LOCATION_FOUND",
                                      }
                    }
                    t={[Function]}
          />
          <Translate(withRouter(Outcomes))
                    currentContactId=""
                    currentIssue={
                              Object {
                                        "categories": Array [],
                                        "id": "1",
                                        "inactive": false,
                                        "link": "",
                                        "linkTitle": "",
                                        "name": "testName",
                                        "outcomeModels": Array [],
                                        "reason": "",
                                        "script": "",
                                      }
                    }
                    numberContactsLeft={0}
                    onSubmitOutcome={[Function]}
                    t={[Function]}
          />
          
</section>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Call
    callState={
        Object {
            "completedIssueIds": Array [
              "test1",
              "test2",
            ],
            "contactIndexes": Object {
              "test1": 2,
              "test2": 1,
            },
            "currentIssueId": "test1",
            "group": undefined,
          }
    }
    clearLocation={[Function]}
    issue={
        Object {
            "categories": Array [],
            "id": "1",
            "inactive": false,
            "link": "",
            "linkTitle": "",
            "name": "testName",
            "outcomeModels": Array [],
            "reason": "",
            "script": "",
          }
    }
    locationState={
        Object {
            "address": "1234",
            "cachedCity": "Anytown",
            "locationFetchType": "BROWSER_GEOLOCATION",
            "splitDistrict": false,
            "uiState": "LOCATION_FOUND",
          }
    }
    onSubmitOutcome={[Function]}
    splitDistrict={false}
    t={[Function]}
/>,
}
`;
